content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+Lastly, let's add a modifier to the speed of the obstacles so that they move a little bit faster as the score increases:
+
+```ts
+
+export class GameApp {
+    static ScrollSpeed : number = 3;
+    ...
+}
+
+class ScrollingObject {
+    sprite: PIXI.Sprite;
+    solid = true;
+
+    public constructor(spriteName: string, isSolid: boolean) {
+        // todo explain
+        this.sprite =  GetSprite(spriteName);
+        this.sprite.y = GameApp.Width;
+        this.sprite.x = GameApp.GroundPosition;
+
+        this.sprite.anchor.set(0, 1);
+        this.solid = isSolid;
+    }
+
+    public Update(delta:number) {
+        var baseScrollSpeed = (this.solid) ? GameApp.ScrollSpeed : GameApp.ScrollSpeed-1;
+
+        // modifier for speed depending on score so that it gets more difficult
+        var scrollSpeed = baseScrollSpeed + Math.min(GameApp.Score/15.0 , 1);
+
+        // move to the left, watch out!
+        this.sprite.x -= delta * (scrollSpeed);
+    }
+}
+```
+
+Right now we can create multiple obstacle objects and have them update on the main game loop. We mentioned before that we wanted to prepare our game for an arbitrary number of objects, how can we achieve this? One way would be to create a list of `WorldObject`s and insert there all the objects that we want.
+
+At this point, however, we actually have 2 types of objects that interact with the world: `Player` and `ScrollingObject`. But it makes sense for us to coalesce all instances in one list of objects to loop over. If we were to go in an [object-oriented](https://en.wikipedia.org/wiki/Object-oriented_programming) direction, we could create a class hierarchy that perhaps could make both our classes inherit from a common one that had all the shared functionality and data. Another good approach would be to create an interface that enforced the methods/fields that they both share. However, since we are using TypeScript and it has a [very powerful type system](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types), we can try to implement one of the lesser pragmatic options by creating a Union type and an alias for it. Remember, we are trying to go for low friction solutions for our little project.
+
+So let's go ahead and implement all this on our `GameApp` class.:
+
+```ts
+// creating an alias for our Union type that we can use
+type WorldObjects = Player | ScrollingObject
+
+export class GameApp {
+  public app: PIXI.Application
+
+  static PressedSpace: boolean = false
+  static Stage: PIXI.Container
+  static ActiveEntities: Array<WorldObjects> = new Array<WorldObjects>()
```

Comment:
I think `ActiveEntitites`, `PressedSpace`, `GameOver` and all things that are variables that are used in the course of single game run are the ones that really confuse me why they are not _instance_ variables.

Reply:
Score, ScoreNextObsstacle too